common/meta/raft-store/src/state_machine/sm.rs

```diff
@@ -1140,21 +1157,118 @@ impl StateMachine {
         }
     }
 
-    #[tracing::instrument(level = "debug", skip(self))]
-    pub async fn lookup_table_id(
+    #[tracing::instrument(level = "debug", skip(self, txn_tree))]
+    fn apply_create_share_cmd(
         &self,
-        db_id: u64,
-        name: &str,
-    ) -> Result<Option<SeqV<TableLookupValue>>, MetaError> {
-        match self.table_lookup().get(
-            &(TableLookupKey {
-                database_id: db_id,
-                table_name: name.to_string(),
-            }),
-        ) {
-            Ok(e) => Ok(e),
-            Err(e) => Err(e.into()),
+        tenant: &str,
+        share_name: &str,
+        txn_tree: &TransactionSledTree,
+    ) -> MetaStorageResult<AppliedState> {
+        let share_lookup_tree = txn_tree.key_space::<ShareLookup>();
+        let share_tree = txn_tree.key_space::<Shares>();
+
+        let share_lookup_key = ShareLookupKey::new(tenant.to_string(), share_name.to_string());
+
+        let seq_share_id = share_lookup_tree.get(&share_lookup_key)?;
+
+        if let Some(u) = seq_share_id {
+            let share_id = u.data.0;
+            let prev = share_tree.get(&share_id)?;
+            return Ok(AppliedState::ShareInfo(Change::nochange_with_id(
+                share_id, prev,
+            )));
         }
+
+        let share_id = self.txn_incr_seq(SEQ_SHARE_ID, txn_tree)?;
+
+        let (prev, _) = self.txn_sub_tree_upsert(
+            &share_lookup_tree,
+            &share_lookup_key,
+            &MatchSeq::Exact(0),
+            Operation::Update(ShareLookupValue(share_id)),
+            None,
+        )?;
+
+        // if it is just created
+        if prev.is_some() {
+            let share_id = prev.unwrap().data;
```

Comment:
If you need to unwrap it, `if let Some(x) = prev {}` would be better, or using `match` to extract the inner data. Avoid using `unwrap()`

Reply:
thanks